src/openms/source/QC/MQExporterHelper.cpp

```diff
@@ -0,0 +1,302 @@
+  }
+  return false;
+}
+
+bool MQExporterHelper::isValid(const std::string& filename)
+{
+  return File::writable(filename);
+}
+
+/**
+  @brief Extract a gene name from a protein description by looking for the substring 'GN='
+*/
+String MQExporterHelper::extractGeneName(const String& prot_description)
+{
+  String gene_name;
+  auto pos_gene = prot_description.find("GN=");
+  // description might not contain a gene name ...
+  if (pos_gene == std::string::npos) return gene_name;
+
+  gene_name = prot_description.substr(pos_gene +3, prot_description.find(" ", pos_gene +3));
+  return gene_name;
+}
+
+MQExporterHelper::MQCommonOutputs::MQCommonOutputs(
+  const OpenMS::Feature& f,
+  const OpenMS::ConsensusMap& cmap,
+  const OpenMS::Size c_feature_number,
+  const std::multimap<OpenMS::String, std::pair<OpenMS::Size, OpenMS::Size>>& UIDs,
+  const OpenMS::ProteinIdentification::Mapping& mp_f,
+  const OpenMS::MSExperiment& exp,
+  const std::map<OpenMS::String,OpenMS::String>& prot_mapper)
+{
+  const OpenMS::PeptideHit* ptr_best_hit; // the best hit referring to score
+  const OpenMS::ConsensusFeature& cf = cmap[c_feature_number];
+  if (MQExporterHelper::hasValidPepID_(f, c_feature_number, UIDs, mp_f))
+  {
+    ptr_best_hit = &f.getPeptideIdentifications()[0].getHits()[0];
+  }
+  else if (MQExporterHelper::hasPeptideIdentifications_(cf))
+  {
+    ptr_best_hit = &cf.getPeptideIdentifications()[0].getHits()[0];
+  }
+  else
+  {
+    return; // no valid PepID; nothing to export
+  }
+
+  const OpenMS::AASequence& pep_seq = ptr_best_hit->getSequence();
+  if (pep_seq.empty())
+  {
+    return; // empty AASequence; nothing to export
+  }
+
+  std::map<OpenMS::String, OpenMS::Size> modifications_temp;
+  if (pep_seq.hasNTerminalModification())
+  {
+    const OpenMS::String& n_terminal_modification = pep_seq.getNTerminalModificationName();
+    modifications_temp.emplace(std::make_pair(n_terminal_modification, 1));
+  }
+  if (pep_seq.hasCTerminalModification())
+  {
+    modifications_temp.emplace(std::make_pair(pep_seq.getCTerminalModificationName(), 1));
+  }
+  for (OpenMS::Size i = 0; i < pep_seq.size(); ++i)
+  {
+    if (pep_seq.getResidue(i).isModified())
+    {
+      ++modifications_temp[pep_seq.getResidue(i).getModification()->getFullId()];
+    }
+  }
+
+  if (modifications_temp.empty())
+  {
+    modifications.str("Unmodified");
+  }
+  else
+  {
+    auto it = modifications_temp.begin();
+    modifications.clear();
+    modifications << it->first;
+    ++it;
+    for (; it != modifications_temp.end(); ++it)
+    {
+      modifications << ";" << it->first;
+    }
+  }
+
+  if (pep_seq.hasNTerminalModification())
+  {
+    const OpenMS::String& n_terminal_modification = pep_seq.getNTerminalModificationName();
+    n_terminal_modification.hasSubstring("Acetyl") ? acetyl = '1' : acetyl = '0'; // Acetyl (Protein N-term)
+  }
+  else
+  {
+    acetyl = '0';
+  }
+  oxidation.clear();
```

Comment:
this may not do what you assume it does :)
And it should not be necessary, since the member has never been touched before.